--- xrdp-0.10.0-beta.1/xrdp/xrdp.ini.o	2024-03-11 13:59:30.562337578 +1100
+++ xrdp-0.10.0-beta.1/xrdp/xrdp.ini	2024-03-11 14:01:38.713716136 +1100
@@ -231,13 +231,13 @@ xrdpvr=true
 ; Some session types such as Xorg and Xvnc start a display server.
 ; Startup command-line parameters for the display server are configured
 ; in sesman.ini. See and configure also sesman.ini.
-[Xorg]
-name=Xorg
-lib=libxup.so
-username=ask
-password=ask
-port=-1
-code=20
+#[Xorg]
+#name=Xorg
+#lib=libxup.so
+#username=ask
+#password=ask
+#port=-1
+#code=20
 
 [Xvnc]
 name=Xvnc
@@ -259,13 +259,13 @@ port=-1
 ; Generic VNC Proxy
 ; Tailor this to specific hosts and VNC instances by specifying an ip
 ; and port and setting a suitable name.
-[vnc-any]
-name=vnc-any
-lib=libvnc.so
-ip=ask
-port=ask5900
-username=na
-password=ask
+#[vnc-any]
+#name=vnc-any
+#lib=libvnc.so
+#ip=ask
+#port=ask5900
+#username=na
+#password=ask
 #pamusername=asksame
 #pampassword=asksame
 #delay_ms=2000
