diff -Naur xrdp-0.4.0/sesman/tools/Makefile xrdp-0.4.0.oden/sesman/tools/Makefile
--- xrdp-0.4.0/sesman/tools/Makefile	2008-05-15 17:29:00.000000000 +0200
+++ xrdp-0.4.0.oden/sesman/tools/Makefile	2008-05-15 17:28:39.000000000 +0200
@@ -14,7 +14,7 @@
 DEFINES = -DLIBSCP_CLIENT
 
 CFLAGS = $(RPM_OPT_FLAGS) -I../../common -I../ -I/usr/include/nptl -I../libscp $(DEFINES)
-LDFLAGS = -L/usr/gnu/lib -L/usr/lib/nptl -L../libscp -Wl,-rpath,. -lpthread -ldl -lscp
+LDFLAGS = -L/usr/gnu/lib -L/usr/lib/nptl -L../libscp -lpthread -ldl -lscp
 C_OS_FLAGS = $(CFLAGS) -c
 LDADD = -ldl -lpthread -lscp
 CC = gcc
diff -Naur xrdp-0.4.0/xrdp/Makefile xrdp-0.4.0.oden/xrdp/Makefile
--- xrdp-0.4.0/xrdp/Makefile	2008-05-15 17:29:00.000000000 +0200
+++ xrdp-0.4.0.oden/xrdp/Makefile	2008-05-15 17:28:52.000000000 +0200
@@ -18,7 +18,7 @@
 CFLAGS = $(RPM_OPT_FLAGS) -I../common -I../libxrdp $(DEFINES)
 #CFLAGS += -DXRDP_DEBUG
 C_OS_FLAGS = $(CFLAGS) -c
-LDFLAGS = -L/usr/gnu/lib -L../libxrdp -Wl,-rpath,.
+LDFLAGS = -L/usr/gnu/lib -L../libxrdp
 LIBS = -ldl -lpthread -lxrdp
 CC = gcc
 
