--- xrdp-0.10.0-beta.1/instfiles/xrdp.service.in.o	2024-03-10 02:06:29.000000000 +1100
+++ xrdp-0.10.0-beta.1/instfiles/xrdp.service.in	2024-03-11 14:10:31.607973918 +1100
@@ -2,7 +2,7 @@
 Description=xrdp daemon
 Documentation=man:xrdp(8) man:xrdp.ini(5)
 Requires=xrdp-sesman.service
-After=network.target xrdp-sesman.service
+After=xrdp-sesman.service
 
 [Service]
 Type=exec
--- xrdp-0.10.0-beta.1/instfiles/xrdp-sesman.service.in.o	2024-03-11 14:14:12.510062559 +1100
+++ xrdp-0.10.0-beta.1/instfiles/xrdp-sesman.service.in	2024-03-11 14:11:27.817255389 +1100
@@ -1,7 +1,8 @@
 [Unit]
 Description=xrdp session manager
 Documentation=man:xrdp-sesman(8) man:sesman.ini(5)
-After=network.target
+Wants=network-online.target nss-lookup.target nss-user-lookup.target
+After=network-online.target nss-lookup.target nss-user-lookup.target
 StopWhenUnneeded=true
 BindsTo=xrdp.service
 
