diff -Naur xrdp-0.4.0/libxrdp/Makefile xrdp-0.4.0.oden/libxrdp/Makefile
--- xrdp-0.4.0/libxrdp/Makefile	2008-06-03 11:35:06.000000000 +0200
+++ xrdp-0.4.0.oden/libxrdp/Makefile	2008-06-03 11:32:23.000000000 +0200
@@ -16,7 +16,7 @@
 CFLAGS = $(RPM_OPT_FLAGS) -I../common -fPIC $(DEFINES)
 #CFLAGS += -DXRDP_DEBUG
 C_OS_FLAGS = $(CFLAGS) -c
-LDFLAGS = -shared
+LDFLAGS = -shared -Wl,-soname=libxrdp.so.0
 LIBS = -ldl
 LIBS += -lcrypto
 CC = gcc
@@ -28,16 +28,17 @@
 	ranlib libxrdp.a
 
 libxrdp: $(LIBXRDPOBJ)
-	$(CC) $(LDFLAGS) -o libxrdp.so $(LIBXRDPOBJ) $(LIBS)
+	$(CC) $(LDFLAGS) -o libxrdp.so.0 $(LIBXRDPOBJ) $(LIBS)
+	ln -snf libxrdp.so.0 libxrdp.so
 
 clean:
-	rm -f $(LIBXRDPOBJ) libxrdp.a libxrdp.so
+	rm -f $(LIBXRDPOBJ) libxrdp.a libxrdp.so.0
 
 install:
-	install libxrdp.so $(DESTDIR)/libxrdp.so
+	install libxrdp.so.0 $(DESTDIR)/libxrdp.so.0
 
 installdeb:
-	install libxrdp.so $(DESTDIRDEB)/usr/lib/xrdp/libxrdp.so
+	install libxrdp.so.0 $(DESTDIRDEB)/usr/lib/libxrdp.so.0
 
 file.o: ../common/file.c
 	$(CC) $(C_OS_FLAGS) ../common/file.c
diff -Naur xrdp-0.4.0/rdp/Makefile xrdp-0.4.0.oden/rdp/Makefile
--- xrdp-0.4.0/rdp/Makefile	2008-06-03 11:35:06.000000000 +0200
+++ xrdp-0.4.0.oden/rdp/Makefile	2008-06-03 11:36:24.000000000 +0200
@@ -13,7 +13,7 @@
 CFLAGS = $(RPM_OPT_FLAGS) -I../common -fPIC $(DEFINES)
 #CFLAGS += -DXRDP_DEBUG
 C_OS_FLAGS = $(CFLAGS) -c
-LDFLAGS = -shared
+LDFLAGS = -shared -Wl,-soname=librdp.so.0
 LIBS = -ldl
 LIBS += -lcrypto
 CC = gcc
@@ -21,10 +21,11 @@
 all: rdp
 
 rdp: $(RDPOBJ)
-	$(CC) $(LDFLAGS) -o librdp.so $(RDPOBJ) $(LIBS)
+	$(CC) $(LDFLAGS) -o librdp.so.0 $(RDPOBJ) $(LIBS)
+	ln -snf librdp.so.0 librdp.so
 
 clean:
-	rm -f $(RDPOBJ) librdp.so
+	rm -f $(RDPOBJ) librdp.so.0
 
 os_calls.o: ../common/os_calls.c
 	$(CC) $(C_OS_FLAGS) ../common/os_calls.c
@@ -33,7 +34,7 @@
 	$(CC) $(C_OS_FLAGS) ../common/ssl_calls.c
 
 install:
-	install librdp.so $(DESTDIR)/librdp.so
+	install librdp.so.0 $(DESTDIR)/librdp.so.0
 
 installdeb:
-	install librdp.so $(DESTDIRDEB)/usr/lib/xrdp/librdp.so
+	install librdp.so.0 $(DESTDIRDEB)/usr/lib/librdp.so.0
diff -Naur xrdp-0.4.0/sesman/libscp/Makefile xrdp-0.4.0.oden/sesman/libscp/Makefile
--- xrdp-0.4.0/sesman/libscp/Makefile	2008-06-03 11:35:06.000000000 +0200
+++ xrdp-0.4.0.oden/sesman/libscp/Makefile	2008-06-03 11:33:11.000000000 +0200
@@ -12,19 +12,20 @@
 
 DESTDIRDEB = /tmp
 
-LIBSCPLNAME = libscp.so
+LIBSCPLNAME = libscp.so.0
 
 DEFINES = -DSESMAN_CFG_FILE=\"$(CFGDIR)/sesman.ini\" \
 	  -DSESMAN_PID_FILE=\"$(PIDDIR)/sesman.pid\" \
 	  -DSESMAN_SESSVC_FILE=\"sessvc\"
 
 CFLAGS = $(RPM_OPT_FLAGS) -I../../common -I/usr/include/nptl -fPIC $(DEFINES)
-LDFLAGS = -shared -L/usr/gnu/lib -L/usr/lib/nptl -lpthread
+LDFLAGS = -shared -Wl,-soname=libscp.so.0 -lpthread
 C_OS_FLAGS = $(CFLAGS) -c
 CC = gcc
 
 all: $(LIBSCPOBJ)
 	$(CC) $(LDFLAGS) -o $(LIBSCPLNAME) $(LIBSCPOBJ)
+	ln -snf $(LIBSCPLNAME) libscp.so
 
 clean:
 	rm -f $(LIBSCPOBJ) $(LIBSCPLNAME)
@@ -33,7 +34,7 @@
 	install $(LIBSCPLNAME) $(DESTDIR)/$(LIBSCPLNAME)
 
 installdeb:
-	install $(LIBSCPLNAME) $(DESTDIRDEB)/usr/lib/xrdp/$(LIBSCPLNAME)
+	install $(LIBSCPLNAME) $(DESTDIRDEB)/usr/lib/$(LIBSCPLNAME)
 
 os_calls.o: ../../common/os_calls.c
 	$(CC) $(C_OS_FLAGS) ../../common/os_calls.c
diff -Naur xrdp-0.4.0/vnc/Makefile xrdp-0.4.0.oden/vnc/Makefile
--- xrdp-0.4.0/vnc/Makefile	2008-06-03 11:35:06.000000000 +0200
+++ xrdp-0.4.0.oden/vnc/Makefile	2008-06-03 11:37:07.000000000 +0200
@@ -12,17 +12,18 @@
 
 CFLAGS = $(RPM_OPT_FLAGS) -I../common -fPIC $(DEFINES)
 C_OS_FLAGS = $(CFLAGS) -c -g
-LDFLAGS = -shared
+LDFLAGS = -shared -Wl,-soname=libvnc.so.0
 LIBS = -ldl
 CC = gcc
 
 all: vnc
 
 vnc: $(VNCOBJ)
-	$(CC) $(LDFLAGS) -o libvnc.so $(VNCOBJ) $(LIBS)
+	$(CC) $(LDFLAGS) -o libvnc.so.0 $(VNCOBJ) $(LIBS)
+	ln -snf libvnc.so.0 libvnc.so
 
 clean:
-	rm -f $(VNCOBJ) libvnc.so
+	rm -f $(VNCOBJ) libvnc.so.0
 
 os_calls.o: ../common/os_calls.c
 	$(CC) $(C_OS_FLAGS) ../common/os_calls.c
@@ -31,7 +32,7 @@
 	$(CC) $(C_OS_FLAGS) ../common/d3des.c
 
 install:
-	install libvnc.so $(DESTDIR)/libvnc.so
+	install libvnc.so.0 $(DESTDIR)/libvnc.so.0
 
 installdeb:
-	install libvnc.so $(DESTDIRDEB)/usr/lib/xrdp/libvnc.so
+	install libvnc.so.0 $(DESTDIRDEB)/usr/lib/libvnc.so.0
diff -Naur xrdp-0.4.0/xup/Makefile xrdp-0.4.0.oden/xup/Makefile
--- xrdp-0.4.0/xup/Makefile	2008-06-03 11:35:06.000000000 +0200
+++ xrdp-0.4.0.oden/xup/Makefile	2008-06-03 11:37:47.000000000 +0200
@@ -13,23 +13,24 @@
 CFLAGS = $(RPM_OPT_FLAGS) -I../common -fPIC $(DEFINES)
 #CFLAGS += -DXRDP_DEBUG
 C_OS_FLAGS = $(CFLAGS) -c
-LDFLAGS = -shared
+LDFLAGS = -shared -Wl,-soname=libxup.so.0
 LIBS = -ldl
 CC = gcc
 
 all: xup
 
 xup: $(XUPOBJ)
-	$(CC) $(LDFLAGS) -o libxup.so $(XUPOBJ) $(LIBS)
+	$(CC) $(LDFLAGS) -o libxup.so.0 $(XUPOBJ) $(LIBS)
+	ln -snf libxup.so.0 libxup.so
 
 clean:
-	rm -f $(XUPOBJ) libxup.so
+	rm -f $(XUPOBJ) libxup.so.0
 
 os_calls.o: ../common/os_calls.c
 	$(CC) $(C_OS_FLAGS) ../common/os_calls.c
 
 install:
-	install libxup.so $(DESTDIR)/libxup.so
+	install libxup.so.0 $(DESTDIR)/libxup.so.0
 
 installdeb:
-	install libxup.so $(DESTDIRDEB)/usr/lib/xrdp/libxup.so
+	install libxup.so.0 $(DESTDIRDEB)/usr/lib/libxup.so.0
