--- sesman/libscp/Makefile.link	2010-04-19 11:48:25.000000000 +0200
+++ sesman/libscp/Makefile	2010-04-19 11:49:01.000000000 +0200
@@ -19,12 +19,12 @@
 	  -DSESMAN_SESSVC_FILE=\"sessvc\"
 
 CFLAGS = $(RPM_OPT_FLAGS) -I../../common -I/usr/include/nptl -fPIC $(DEFINES)
-LDFLAGS = -shared -Wl,-soname=libscp.so.0 -lpthread
+LDFLAGS = -shared -Wl,-soname=libscp.so.0
 C_OS_FLAGS = $(CFLAGS) -c
 CC = gcc
 
 all: $(LIBSCPOBJ)
-	$(CC) $(LDFLAGS) -o $(LIBSCPLNAME) $(LIBSCPOBJ)
+	$(CC) $(LDFLAGS) -o $(LIBSCPLNAME) $(LIBSCPOBJ) -ldl -lpthread
 	ln -snf $(LIBSCPLNAME) libscp.so
 
 clean:
--- libxrdp/Makefile.link	2010-04-19 11:58:37.000000000 +0200
+++ libxrdp/Makefile	2010-04-19 11:59:03.000000000 +0200
@@ -2,7 +2,7 @@
 LIBXRDPOBJ = libxrdp.o xrdp_tcp.o xrdp_iso.o xrdp_mcs.o \
              xrdp_sec.o xrdp_rdp.o xrdp_orders.o \
              xrdp_bitmap_compress.o xrdp_channel.o \
-             os_calls.o ssl_calls.o file.o
+             os_calls.o ssl_calls.o file.o list.o
 
 DESTDIR = /usr/local/xrdp
 CFGDIR = /etc/xrdp
@@ -40,6 +40,9 @@
 installdeb:
 	install libxrdp.so.0 $(DESTDIRDEB)/usr/lib/libxrdp.so.0
 
+list.o: ../common/list.c
+	$(CC) $(C_OS_FLAGS) ../common/list.c
+
 file.o: ../common/file.c
 	$(CC) $(C_OS_FLAGS) ../common/file.c
 
--- sesman/Makefile.link	2010-04-19 12:01:36.000000000 +0200
+++ sesman/Makefile	2010-04-19 12:04:38.000000000 +0200
@@ -21,7 +21,7 @@
 CFLAGS = $(RPM_OPT_FLAGS) -I../common -I/usr/include/nptl -I./libscp $(DEFINES) 
 LDFLAGS = -L/usr/gnu/lib -L/usr/lib/nptl -L./libscp
 C_OS_FLAGS = $(CFLAGS) -c
-LDADD = -ldl -lpthread -lscp
+LDADD = -ldl -lpthread
 CC = gcc
 
 all: libscp_ pam tools 
--- sesman/tools/Makefile.link	2010-04-19 12:19:08.000000000 +0200
+++ sesman/tools/Makefile	2010-04-19 12:20:55.000000000 +0200
@@ -3,7 +3,7 @@
 	     os_calls.o
 
 SESRUNOBJ = sesrun.o config.o tcp.o \
-	    os_calls.o d3des.o list.o file.o log.o
+	    os_calls.o d3des.o file.o log.o
 
 DESTDIR = /usr/local/xrdp
 CFGDIR = /etc/xrdp
@@ -14,25 +14,22 @@
 DEFINES = -DLIBSCP_CLIENT
 
 CFLAGS = $(RPM_OPT_FLAGS) -I../../common -I../ -I/usr/include/nptl -I../libscp $(DEFINES)
-LDFLAGS = -L/usr/gnu/lib -L/usr/lib/nptl -L../libscp -lpthread -ldl -lscp
+LDFLAGS = -L/usr/gnu/lib -L/usr/lib/nptl -L../libscp -L../../libxrdp
 C_OS_FLAGS = $(CFLAGS) -c
-LDADD = -ldl -lpthread -lscp
+LDADD = -ldl -lpthread
 CC = gcc
 
 all: stest srun
 
 stest: $(SESTESTOBJ)
-	$(CC) $(LDFLAGS) -o sestest $(SESTESTOBJ) $(LDADD)
+	$(CC) $(LDFLAGS) -o sestest $(SESTESTOBJ) $(LDADD) -lscp
 
 srun: $(SESRUNOBJ)
-	$(CC) $(LDFLAGS) -o sesrun $(SESRUNOBJ) $(LDADD)
+	$(CC) $(LDFLAGS) -o sesrun $(SESRUNOBJ) $(LDADD) -lxrdp
 
 os_calls.o: ../../common/os_calls.c
 	$(CC) $(C_OS_FLAGS) ../../common/os_calls.c
 
-list.o: ../../common/list.c
-	$(CC) $(C_OS_FLAGS) ../../common/list.c
-
 file.o: ../../common/file.c
 	$(CC) $(C_OS_FLAGS) ../../common/file.c
 
