diff -up xrdp-0.10.4/instfiles/xrdp-sesman.service.in.3~ xrdp-0.10.4/instfiles/xrdp-sesman.service.in
--- xrdp-0.10.4/instfiles/xrdp-sesman.service.in.3~	2025-07-02 04:22:36.000000000 +0200
+++ xrdp-0.10.4/instfiles/xrdp-sesman.service.in	2025-07-07 00:18:40.044006130 +0200
@@ -1,7 +1,8 @@
 [Unit]
 Description=xrdp session manager
 Documentation=man:xrdp-sesman(8) man:sesman.ini(5)
-After=network.target
+Wants=network-online.target nss-lookup.target nss-user-lookup.target
+After=network-online.target nss-lookup.target nss-user-lookup.target
 StopWhenUnneeded=true
 BindsTo=xrdp.service
 
diff -up xrdp-0.10.4/instfiles/xrdp.service.in.3~ xrdp-0.10.4/instfiles/xrdp.service.in
--- xrdp-0.10.4/instfiles/xrdp.service.in.3~	2025-07-02 04:22:36.000000000 +0200
+++ xrdp-0.10.4/instfiles/xrdp.service.in	2025-07-07 00:33:58.685953673 +0200
@@ -2,7 +2,7 @@
 Description=xrdp daemon
 Documentation=man:xrdp(8) man:xrdp.ini(5)
 Requires=xrdp-sesman.service
-After=network-online.target xrdp-sesman.service
+After=xrdp-sesman.service
 
 [Service]
 Type=exec
